### PR TITLE
Add options for gate fusion.

### PR DESCRIPTION
**Pull request by sergeisakov**

### CODE REVIEW

File: lib/fuser_basic.h [hunk 1]
```diff
@@ -31,26 +31,35 @@ template <typename IO, typename Gate>
 struct BasicGateFuser final {
   using GateFused = qsim::GateFused<Gate>;
 
+  /**
+   * User-specified parameters for gate fusion.
+   * BasicGateFuser does not use any parameters.
+   */
+  struct Parameter {};
```
Comment (95-martin-orion) [inline lib/fuser_basic.h:38]: I assume fusion with more than two qubits will be handled by a new Fuser type?

**Review by 95-martin-orion — APPROVED** [review]
My reading of this is that users will pass Fuser params into run_qsim calls, and those params will be consumed by a new Fuser implementation, to be defined in a follow-up PR.

Is that correct?

**Comment by sergeisakov** [comment]
> My reading of this is that users will pass Fuser params into run_qsim calls, and those params will be consumed by a new Fuser implementation, to be defined in a follow-up PR.
> 
> Is that correct?

Yes, that is correct.

File: lib/fuser_basic.h (hunk 1 above)
Comment (sergeisakov) [reply lib/fuser_basic.h:38]: Yes, exactly.

**Pull request merged**